vendor.conf

```
@@ -1,5 +1,5 @@
 github.com/Azure/go-ansiterm                        d6e3b3328b783f23731bc4d058875b0371ff8109
-github.com/Microsoft/hcsshim                        5bc557dd210ff2caf615e6e22d398123de77fc11 # v0.8.9
+github.com/Microsoft/hcsshim                        9dcb42f100215f8d375b4a9265e5bba009217a85
```

full diff is https://github.com/microsoft/hcsshim/compare/v0.8.9...9dcb42f100215f8d375b4a9265e5bba009217a85


perhaps would be good to add a comment with `git describe` output, to give a bit of context what version we're vendoring;

```suggestion
github.com/Microsoft/hcsshim                        9dcb42f100215f8d375b4a9265e5bba009217a85 # v0.8.9-1-g9dcb42f
```

---

Oh, actually, I see this is vendored from the `moby` branch?  I see https://github.com/microsoft/hcsshim/commit/9dcb42f100215f8d375b4a9265e5bba009217a85 was cherry-picked from https://github.com/microsoft/hcsshim/pull/832 (which was against master)

In that case we should at least have a comment for that;

```suggestion
github.com/Microsoft/hcsshim                        9dcb42f100215f8d375b4a9265e5bba009217a85 # moby branch
```


Will this cause issues if we update containerd vendoring (which does not vendor from that branch?); are there breaking changes in master that we cannot vendor? I'm a bit concerned if we start vendoring from custom branches as that would definitely complicate things for users using go modules (and if we perhaps one day make that transition as well)

---

I should have given more context on this. :)

We changed to the moby branch for a couple of reasons:
- Allows us to take this important change without needing to also pull in all of the other work that has been going on in the repo.
- moby uses an older set of APIs exposed from hcsshim, based on the HCS v1 functionality. Going forwards, we have discussed deprecating/removing these APIs from the mainline branch in hcsshim, so our thinking was we could keep this moby branch around to ensure we don't break compatibility there.

I can definitely add in a `# moby branch` comment, so it's clear where this is coming from.

Do you know what hcsshim functionality moby uses through containerd package imports? I'm not aware of anything, but that is something we would need to be careful of with this approach. Long term, the best path here is to get moby using containerd as the backend on Windows, which should alleviate these issues.

---

This is fine for now. We'll want to migrate windows to use containerd and we'll need to be careful to update this vendoring (assuming we need it even) once that is done.